File: app/src/main/java/com/github/malitsplus/shizurunotes/ui/clanbattle/ClanBattleFragment.java
package com.github.malitsplus.shizurunotes.ui.clanbattle;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.lifecycle.Observer;
import androidx.lifecycle.ViewModelProviders;

import com.github.malitsplus.shizurunotes.R;

public class ClanBattleFragment extends Fragment {

    private ClanBattleViewModel clanBattleViewModel;

    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container, Bundle savedInstanceState) {
        clanBattleViewModel =
                ViewModelProviders.of(this).get(ClanBattleViewModel.class);
        View root = inflater.inflate(R.layout.fragment_clan_battle, container, false);



        /*
        final TextView textView = root.findViewById(R.id.text_gallery);
        clanBattleViewModel.getText().observe(this, new Observer<String>() {
            @Override
            public void onChanged(@Nullable String s) {
                textView.setText(s);
            }
        });

         */
        return root;
    }
}

File: app/src/main/java/com/github/malitsplus/shizurunotes/ui/clanbattle/ClanBattleFragment.kt
package com.github.malitsplus.shizurunotes.ui.clanbattle

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.fragment.app.Fragment
import androidx.lifecycle.Observer
import androidx.lifecycle.ViewModelProvider
import androidx.recyclerview.widget.LinearLayoutManager
import com.github.malitsplus.shizurunotes.R
import com.github.malitsplus.shizurunotes.databinding.FragmentClanBattleBinding
import com.github.malitsplus.shizurunotes.ui.shared.SharedViewModelClanBattle
import com.github.malitsplus.shizurunotes.ui.shared.SharedViewModelClanBattleFactory

class ClanBattleFragment : Fragment() {

    private lateinit var clanBattleViewModel: ClanBattleViewModel
    private lateinit var sharedClanBattle: SharedViewModelClanBattle
    private lateinit var adapter: ClanBattleAdapter

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedClanBattle = ViewModelProvider(activity!!)[SharedViewModelClanBattle::class.java]
        clanBattleViewModel = ViewModelProvider(this, SharedViewModelClanBattleFactory(sharedClanBattle))[ClanBattleViewModel::class.java]
    }

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {

        val binding =
            DataBindingUtil.inflate<FragmentClanBattleBinding>(
                inflater, R.layout.fragment_clan_battle, container, false
            ).apply{
                lifecycleOwner = viewLifecycleOwner
                adapter = ClanBattleAdapter(context!!, sharedClanBattle)
                clanBattleListRecycler.layoutManager = LinearLayoutManager(context)
                clanBattleListRecycler.adapter = adapter
                clanBattleListRecycler.setHasFixedSize(true)
            }

        sharedClanBattle.apply {
            periodList.observe(
                viewLifecycleOwner, Observer{
                    adapter.update(it)
                }
            )
            loadingFlag.observe(
                viewLifecycleOwner, Observer {
                    if (it) binding.clanBattleProgressBar.visibility = View.VISIBLE
                    else binding.clanBattleProgressBar.visibility = View.GONE
                }
            )
        }

        binding.viewModel = clanBattleViewModel
        sharedClanBattle.loadData()
        return binding.root
    }
}